tests/torchtune/utils/test_checkpoint.py

```diff
@@ -1,138 +0,0 @@
-        for p in model.parameters():
-            p.grad = torch.rand_like(p)
-        optim.step()
-        checkpoint = {"model": model, "optimizer": optim, "lr": 0.01}
-        model_new, optim_new = self._get_model_and_optim(zero_model=True, fsdp=False)
-        # Saves checkpoint, calls load_checkpoint and loads model + optim states into new model/optim.
-        self._save_and_load(checkpoint, model_new, optim_new)
-        # model_new and model params should match
-        for p1, p2 in zip(model.parameters(), model_new.parameters()):
-            torch.testing.assert_close(p1, p2)
-        # optim state_dicts should match
-        self._validate_dicts(optim.state_dict(), optim_new.state_dict())
-
-    def test_no_model_key_save(self) -> None:
-        checkpoint = {"lr": 0.03}
-        with pytest.raises(
-            RuntimeError,
-            match="Expected `ckpt_dict` to contain a `model` key, but it does not.",
-        ):
-            with tempfile.NamedTemporaryFile() as f:
-                save_checkpoint(checkpoint, f.name)
-
-    def test_no_model_key_load(self) -> None:
-        model = torch.nn.Linear(1, 1)
-        with tempfile.NamedTemporaryFile() as f:
-            torch.save({"lr": 0.01}, f.name)
-            with pytest.raises(
-                RuntimeError,
-                match="Expected loaded checkpoint to contain a `model` key.*",
-            ):
-                load_checkpoint(f.name, model)
-
-    def test_no_optim_key_load(self) -> None:
-        model = torch.nn.Linear(1, 1)
-        optim = torch.optim.SGD(model.parameters(), lr=0.01)
-        with tempfile.NamedTemporaryFile() as f:
-            save_checkpoint({"model": model}, f.name)
-            with pytest.raises(
-                RuntimeError,
-                match="Expected loaded checkpoint to contain an `optimizer` key.*",
-            ):
-                load_checkpoint(f.name, model, optim)
-
-    def _test_distributed_save_load(self) -> None:
-        torch.distributed.init_process_group(backend="nccl")
-        torch.cuda.set_device(torch.distributed.get_rank())
-        torch.distributed.barrier()
-        model, optim = self._get_model_and_optim(zero_model=False, fsdp=True)
-        for p in model.parameters():
-            p.grad = torch.rand_like(p)
-        optim.step()
-        checkpoint = {"model": model, "optimizer": optim, "lr": 0.01}
-        model_new, optim_new = self._get_model_and_optim(zero_model=True, fsdp=True)
-        # Saves checkpoint, calls load_checkpoint and loads model + optim states into new model/optim.
-        self._save_and_load(checkpoint, model_new, optim_new)
-
-        # Verify  model
-        with FSDP.summon_full_params(model_new):
-            with FSDP.summon_full_params(model):
-                for p1, p2 in zip(model.parameters(), model_new.parameters()):
-                    torch.testing.assert_close(p1, p2)
-        # Verify optim state_dicts
-        self._validate_dicts(
```

Comment:
I think this eliminates our testing coverage around FSDP optimizer state checkpointing?